subprojects/shipkit/src/test/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFileTest.groovy

```diff
@@ -0,0 +1,74 @@
+package org.shipkit.internal.gradle.java.tasks
+
+import org.gradle.testfixtures.ProjectBuilder
+import testutil.PluginSpecification
+
+class CreateDependencyInfoFileTest extends PluginSpecification {
+
+    CreateDependencyInfoFile createDependencyInfoFile = new CreateDependencyInfoFile()
+
+    def "should create correct output file"() {
+        given:
+
+        def child = new ProjectBuilder().withName("child").withParent(project).build()
+        child.setVersion("1.2.3")
+        child.setGroup("projGroup")
+        def sibling = new ProjectBuilder().withName("sibling").withParent(project).build()
+        sibling.setVersion("1.2.3")
+        sibling.setGroup("projGroup")
+
+        child.repositories.jcenter()
+        child.configurations.create("runtime")
+        child.dependencies {
+            runtime child.project(":sibling")
+            runtime "projGroup:sibling:1.1.1"
+            runtime "junit:junit:4.10"
+            runtime "org.mockito:mockito-core:2.10.0:sources"
+            runtime "org.mockito:mockito-core:2.10.0"
+        }
+
+        def runtimeConf = child.configurations.getByName("runtime")
+
+        def createDependencyInfoFileTask = project.tasks.create("createTask", CreateDependencyInfoFileTask)
+
+        def output = tmp.newFile("output")
+        createDependencyInfoFileTask.outputFile = output
+        createDependencyInfoFileTask.configuration = runtimeConf
+        createDependencyInfoFileTask.projectGroup = "projGroup"
+        createDependencyInfoFileTask.currentProjectVersion = "1.2.3"
+
+        when:
+        createDependencyInfoFile.createDependencyInfoFile(createDependencyInfoFileTask)
+
+        then:
+        output.text == """{
+\t"description":"This file was generated by Shipkit Gradle plugin. See http:\\/\\/shipkit.org\\n\\n",
+\t"dependencies":[
+\t\t{
+\t\t\t"dependencyGroup":"projGroup",
```

Comment:
How about we dedupe the metadata and make it a bit more structured?

```JSON
{
    "description":"This file was...",
    "dependencies":[
        {
            "group": "foo",
            "name": "bar",
            "submodule": false,
            "version": "1.0",
            "artifact": {}
        },
        {
            "group": "foo",
            "name": "bar",
            "submodule": true,
            "version": null,
            "artifact": {}
        },        
        {
            "group": "foo",
            "name": "bar",
            "submodule": false,
            "version": "1.0",
            "artifact": {
                "type": "jar",
                "name": "mockito-core",
                "classifier": "sources",
                "extension": "jar"
            }
        }
    ]
}
```

I was thinking whether it is better to do ```"version": null``` or to skip the version from metadata. Same for "artifact". I'm leaning towards keeping the keys with empty/null values. This way it is more explicit. The users when they see it they can build code around it, rather find out in surprising way when code breaks.

Reply:
OK, I agree, let's keep the null values.